### PR TITLE
Added Branch Coverage Tool & Unit Tests for compare()

### PR DESCRIPTION
**Pull request by SoFoDa**
#### Added
* Implemented ad-hoc branch coverage tool for EntryComparator::compare().
* Implemented 4 unit tests that increase the coverage from 28% to 48% according to our own branch coverage tool.

### CODE REVIEW

**Review by christinerosquist — APPROVED** [review]

File: src/main/java/org/jabref/logic/bibtex/comparator/EntryComparator.java [hunk 1]
```diff
@@ -43,86 +49,148 @@ public int compare(BibEntry e1, BibEntry e2) {
...
             }
         } else if (sortField.equals(InternalField.TYPE_HEADER)) {
+            visited[11] = true;           
             // Sort by type.
             f1 = e1.getType();
             f2 = e2.getType();
         } else if (sortField.isNumeric()) {
+            visited[12] = true;           
             try {
                 Integer i1 = Integer.parseInt((String) f1);
```
Comment (simonsirak) [inline src/main/java/org/jabref/logic/bibtex/comparator/EntryComparator.java:103]: Nice, you got to handle a try/catch! 😄

File: src/test/java/org/jabref/logic/bibtex/comparator/EntryComparatorTest.java [hunk 2]
```diff
@@ -14,4 +15,70 @@ void recognizeIdenticObjectsAsEqual() {
...
+     */
+    @Test
+    void testEntryComparator2() {
+        BibEntry e1 = new BibEntry(StandardEntryType.Article);        
+        e1.setField(StandardField.AUTHOR, "ABC ABB");        
+        
+        BibEntry e2 = new BibEntry(StandardEntryType.Article);        
+        e2.setField(StandardField.AUTHOR, "ABB ABC");        
+
+        assertEquals(1, new EntryComparator(false, false, StandardField.AUTHOR).compare(e1, e2));
```
Comment (simonsirak) [inline src/test/java/org/jabref/logic/bibtex/comparator/EntryComparatorTest.java:47]: Does 1 mean that `e1 > e2`? What is the surname in this case, is it `e1->ABB` and `e2->ABC`? I might've misunderstood something here!

**Review by simonsirak — APPROVED** [review]
Looks great, good job! 😄 I like that these tests are focused/concise, they are very easy to read and understand!

File: src/test/java/org/jabref/logic/bibtex/comparator/EntryComparatorTest.java (hunk 2 above)
Comment (SoFoDa) [reply src/test/java/org/jabref/logic/bibtex/comparator/EntryComparatorTest.java:47]: Looking into it more, I think the function interprets the first word as the surname. As if the author's name would be written as "Knuth, Donald" which results in `e1 > e2`.

**Pull request merged**